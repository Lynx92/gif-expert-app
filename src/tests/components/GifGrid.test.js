import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("GifGrid", () => {
  const gifs = [
    {
      id: 1,
      url: "http://www.w3.org/1999/",
      title: "cosa",
    },
  ];

  useFetchGifs.mockReturnValue({
    data: gifs,
    loading: false,
  });

  const category = "Mola";
  const wrapper = shallow(<GifGrid category={category} />);

  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should not show loading", () => {
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
