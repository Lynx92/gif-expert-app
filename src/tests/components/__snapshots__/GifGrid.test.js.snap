// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should render 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Mola
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="cosa"
      url="http://www.w3.org/1999/"
    />
  </div>
</Fragment>
`;
